apps/condo/domains/common/utils/text.js

```diff
@@ -10,8 +17,8 @@ function normalizeText (text) {
         // replace two or more spaces to one space
         .replace(/\p{Z}+/gu, ' ')
         // normalize punctuation between words, e.g: 'test  ,test' -> 'test, test'
-        .replace(new RegExp(`[\\p{L}\\p{N}] *[${punctuations}]+ *`, 'gmu'), wordWithPunctuation => (
-            `${wordWithPunctuation.replace(new RegExp(` *[${punctuations}] *`, 'gm'), punctuation => punctuation.trim())} `
+        .replace(PUNCTUATION_SEARCH_REGEX, wordWithPunctuation => (
+            `${wordWithPunctuation.replace(PUNCTUATION_PART_REGEX, punctuation => punctuation.trim())} `
```

Comment:
Is last space char a typo here?

Reply:
No, it is used to replace '    ,   ' to trimmed char and spacee ',' + ' ' = ', '